src/files/XmlFile.ts

```diff
@@ -489,7 +489,7 @@ export class XmlFile {
 
         if (this.needsTranspiled || extraImportScripts.length > 0) {
             //temporarily add the missing imports as script tags
-            const originalScripts = this.ast.component.scripts;
+            const originalScripts = this.ast.component?.scripts ?? [];
```

Comment:
Unrelated to this PR. Just a little extra safeguarding around potentially-broken component AST.